extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsTemplate.java

```diff
@@ -42,31 +41,6 @@
     private static final String MEMORY_NON_HEAP_USAGE = "memory.nonHeap.usage";
     private static final String THREAD_COUNT = "thread.count";
 
-    /*
-     * public <E extends Member & AnnotatedElement> void registerCounted(Class<?> topClass, E element) {
-     * MetricRegistry registry = MetricRegistries.get(MetricRegistry.Type.APPLICATION);
-     * //registry.register(name, new CounterImpl());
-     * MetricResolver resolver = new MetricResolver();
-     * MetricResolver.Of<Counted> of = resolver.counted(topClass, element);
-     * Metadata meta = new Metadata(of.metricName(), MetricType.COUNTER);
-     * //registry.register(meta, new CounterImpl());
-     * }
-     */
-    public void registerCounted(String topClassName, String elementName, ShutdownContext shutdown) {
-        MetricRegistry registry = MetricRegistries.get(MetricRegistry.Type.APPLICATION);
-
-        String name = MetricRegistry.name(topClassName, elementName);
-        Metadata meta = new Metadata(name, MetricType.COUNTER);
-        log.debugf("Register: %s", name);
-        registry.register(meta, new CounterImpl());
-        shutdown.addShutdownTask(new Runnable() {
-            @Override
-            public void run() {
-                registry.remove(name);
-            }
-        });
-    }
-
```

Comment:
We can address this comment in a follow-up PR. Let's get this one committed first as we have people asking for it.

Could you check why we have some metrics commented below? I suspect one of the reason is the usage of the MX pool but maybe we should use `ImageInfo` to disable them only if we are in native image mode.

I don't know about the CPU one though.

Having comments would help.

Reply:
I will look into it, reported https://github.com/quarkusio/quarkus/issues/1419 I don't have the permission to assign it to myself though.